Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp

```diff
@@ -1186,6 +1198,12 @@ size_t LineBuilder::rebuildLineWithInlineContent(const InlineItemRange& layoutRa
         if (&inlineItem == &lastInlineItemToAdd)
             break;
     }
+    // Remove floats that are outside of this "rebuild" range to ensure we don't add them twice.
+    for (; index < layoutRange.endIndex(); ++index) {
+        if (m_inlineItemList[index].isFloat())
+            unplaceFloatBox(m_inlineItemList[index].layoutBox());
```

Comment:
undoing float placement is very specific to this "rebuild line" use case so I think it'd better if this functionality stayed private to this function. 
I think it's fine to have a simple `PlacedFloats::remove(const Box&) `and just call `removeFirst`.
-also this loop can bail out early on the first float that has not been placed yet.

Reply:
All comments have been fixed.